### ISSUE TITLE
test_bitcoin fails doing ./qa/zcash/check-security-hardening.sh with 1.0.1, 1.0.2, and 1.0.3

### ISSUE THREAD

**Issue by aayanl** [z-classic/zclassic#28]
Error message:
```FAIL: test/test_bitcoin

Running 197 test cases...
test/alert_tests.cpp(291): error: in "Alert_tests/AlertApplies": check alert.CheckSignature(alertKey) has failed
test/alert_tests.cpp(291): error: in "Alert_tests/AlertApplies": check alert.CheckSignature(alertKey) has failed
test/alert_tests.cpp(291): error: in "Alert_tests/AlertApplies": check alert.CheckSignature(alertKey) has failed
test/alert_tests.cpp(291): error: in "Alert_tests/AlertApplies": check alert.CheckSignature(alertKey) has failed
test/alert_tests.cpp(291): error: in "Alert_tests/AlertApplies": check alert.CheckSignature(alertKey) has failed
test/alert_tests.cpp(291): error: in "Alert_tests/AlertApplies": check alert.CheckSignature(alertKey) has failed
test/alert_tests.cpp(291): error: in "Alert_tests/AlertApplies": check alert.CheckSignature(alertKey) has failed
test/alert_tests.cpp(291): error: in "Alert_tests/AlertApplies": check alert.CheckSignature(alertKey) has failed
test/alert_tests.cpp(291): error: in "Alert_tests/AlertApplies": check alert.CheckSignature(alertKey) has failed
test/alert_tests.cpp(291): error: in "Alert_tests/AlertApplies": check alert.CheckSignature(alertKey) has failed
test/alert_tests.cpp(297): error: in "Alert_tests/AlertApplies": check alerts[0].AppliesTo(1, "") has failed
test/alert_tests.cpp(298): error: in "Alert_tests/AlertApplies": check alerts[0].AppliesTo(999001, "") has failed
test/alert_tests.cpp(299): error: in "Alert_tests/AlertApplies": check alerts[0].AppliesTo(1, "/MagicBean:11.11.11/") has failed
test/alert_tests.cpp(301): error: in "Alert_tests/AlertApplies": check alerts[1].AppliesTo(1, "/MagicBean:0.1.0/") has failed
test/alert_tests.cpp(302): error: in "Alert_tests/AlertApplies": check alerts[1].AppliesTo(999001, "/MagicBean:0.1.0/") has failed
test/alert_tests.cpp(304): error: in "Alert_tests/AlertApplies": check alerts[2].AppliesTo(1, "/MagicBean:0.1.0/") has failed
test/alert_tests.cpp(305): error: in "Alert_tests/AlertApplies": check alerts[2].AppliesTo(1, "/MagicBean:0.2.0/") has failed
test/alert_tests.cpp(339): error: in "Alert_tests/AlertNotify": check r.size() == 6u has failed [0 != 6]
unknown location(0): fatal error: in "Alert_tests/AlertNotify": memory access violation at address: 0x00000008: no mapping at fault address
test/alert_tests.cpp(345): last checkpoint

*** 19 failures are detected in the test module "Bitcoin Test Suite"
test_bitcoin: /home/aayanl/zcl1.0.3/depends/x86_64-unknown-linux-gnu/share/../include/boost/thread/pthread/condition_variable_fwd.hpp:102: boost::condition_variable::~condition_variable(): Assertion `!ret' failed.
FAIL test/test_bitcoin (exit status: 134)
```

**Comment by hairetikos** [comment]
What test did you invoke there?  Seems like a memory error you have this time.  We were seeing different errors before:

$ /qa/zcash/check-security-hardening.sh
...
../build-aux/test-driver: line 107: 6101 Aborted "$@" > $log_file 2>&1
FAIL: test/test_bitcoin

commented out line 107 of test-driver:
...
PASS: test/test_bitcoin

but this is not the solution.

**Comment by aayanl** [comment]
@hairetikos I'm doing full-test-suite.sh. I tried with 6, 8, and 10 gigs of ram and it didn't solve the issue. Commenting out line 107 made the test pass, but as you said, it isn't the solution

**Comment by hairetikos** [comment]
@aayanl i realise now the more detailed log you pasted shows after leaving the test for some time after it has failed and "Aborted".  Hopefully we can figure this out with the info

**Comment by hairetikos** [comment]
@mountaindrifter has found this gem, we think it may be very related as the first error which triggers the escalation is "CheckSignature(alertKey) has failed"

https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2016-September/013104.html

and from alerts_test.cpp:

[9:09]  
* If the alert key pairs have changed, the test suite will fail as the
* test data is now invalid.  To create valid test data, signed with a
* new alert private key, follow these steps:
*
* 1. Copy your private key into alertkeys.h.  Don't commit this file!
*    See sendalert.cpp for more info.
*
* 2. Set the GENERATE_ALERTS_FLAG to true.
*
* 3. Build and run:
*    test_bitcoin -t Generate_Alert_Test_Data
*
* 4. Test data is saved in your current directory as alertTests.raw.NEW
*    Copy this file to: src/test/data/alertTests.raw
*
*    For debugging purposes, terminal output can be copied into:
*    src/test/data/alertTests.raw.h
*
* 5. Clean up...
*    - Set GENERATE_ALERTS_FLAG back to false.
*    - Remove your private key from alertkeys.h
*
* 6. Build and verify the new test data:
*    test_bitcoin -t Alert_tests
*
*/

**Comment by hairetikos** [comment]
me and @mountaindrifter have ironed out the issues to do with CheckSignature(alertKey), i have removed a lot of code to do with the sendalert & key system

https://bitcoin.org/en/alert/2016-11-01-alert-retirement

the test fails *Gracefully* now due to a separate issue:

test/main_tests.cpp(62): error: in "main_tests/subsidy_limit_test": check nSum == 12500000000000ULL has failed [1250000000 != 12500000000000]

this is to do with the difference in slow start from zcash to zclassic -- it is an artifact not changed to match zclassic:

/ Mining slow start
   for (int nHeight = 0; nHeight < consensusParams.nSubsidySlowStartInterval; nHeight ++) {
       CAmount nSubsidy = GetBlockSubsidy(nHeight, consensusParams);
       BOOST_CHECK(nSubsidy <= 12.5 * COIN);
       nSum += nSubsidy;
       BOOST_CHECK(MoneyRange(nSum));
   }
   BOOST_CHECK_EQUAL(nSum, 12500000000000ULL);

@heyrhett set slow start to 10 blocks:

https://github.com/z-classic/zclassic/commit/0677deee3774d65893090e126b80418bd026cb60

consensus.nSubsidySlowStartInterval = 20000;         +        consensus.nSubsidySlowStartInterval = 10

we need to change the test to match the zclassic values.

opening a new issue regarding this, as it is something else unrelated to previous.

**Comment by hairetikos** [comment]
https://github.com/z-classic/zclassic/issues/29

**Comment by aayanl** [comment]
Has been fixed with 1.0.3
